source/hpack.c

```diff
@@ -691,16 +700,18 @@ int aws_hpack_insert_header(struct aws_hpack_context *context, const struct aws_
     /* allocate memory for the name and value, which will be deallocated whenever the entry is evicted from the table or
      * the table is cleaned up. We keep the pointer in the name pointer of each entry */
     const size_t buf_memory_size = header->name.len + header->value.len;
-    uint8_t *buf_memory = aws_mem_acquire(context->allocator, buf_memory_size);
-    if (!buf_memory) {
-        return AWS_OP_ERR;
+    /* if buf_memory_size is 0, no memory needed, we will insert the empty header into dynamic table */
+    if (buf_memory_size) {
+        uint8_t *buf_memory = aws_mem_acquire(context->allocator, buf_memory_size);
+        if (!buf_memory) {
+            return AWS_OP_ERR;
+        }
+        struct aws_byte_buf buf = aws_byte_buf_from_empty_array(buf_memory, buf_memory_size);
+        /* Copy header, then backup strings into our own allocation */
+        *table_header = *header;
```

Comment:
we're not setting `table_header`'s members to anything in the else-0-memory-size case, and I'm not positive that s_dynamic_table_get() returns zeroed-memory in all possible cases. Let's explicitly set its name/value/compression in an else block